### ISSUE TITLE
Support disabling the creation of Request-Context header via configuration

### ISSUE THREAD

**Issue by juho-hanhimaki** [Microsoft/ApplicationInsights-aspnetcore#613]
## Repro Steps
1. Create ASP.NET Core project and enable Application Insights.

##  Actual Behavior
Request-Context HTTP header gets set for every (except first?) request.

## Expected Behavior
Ability to disable the creation of Request-Context header via configuration or other method.

## Version Info 
SDK Version : 2.2.1
.NET Version: 2.0.5

## Workaround
Custom middleware that removes the header.
```
            app.Use((context, next) =>
            {
                context.Response.Headers.Remove("Request-Context");
                return next();
            });
```

## Motivation
APIs for Industrial Internet and other scenarios where the bandwidth and client resources are limited and should be used sparingly.

**Comment by cijothomas** [comment]
https://github.com/Microsoft/ApplicationInsights-aspnetcore/wiki/Dependency-Tracking-and-Performance-Counter-Collection#configuring-telemetry-module-services - Please let me know if this wiki information is useful and can help you.

**Comment by cijothomas** [comment]
Sorry I misread the issue.
As of now there are no config options to achieve this. We are working on config story for aspnet core sdk, and will take this one as part of it.

**Comment by cijothomas** [comment]
This was accidently closed. Need to be addressed along with this:
https://github.com/Microsoft/ApplicationInsights-aspnetcore/issues/650

**Comment by lmolkova** [comment]
Fixed in #716, set `ApplicationInsightsServiceOptions.RequestCollectionOptions.InjectResponseHeaders` to false to disable response header injection.
